### DISCUSSION TITLE
Facing issue with rewrite when trying to resolve nested path names

### DISCUSSION THREAD

**Discussion by icybergenome** [vercel/next.js#44561] [Help]
We are trying to rewrite the path `/account/:account_id/showcase/:showcase_id*` to `/a/:account_id/s/:showcase_id*`. But its not working, is it due to: no support for nested [path matching](https://nextjs.org/docs/api-reference/next.config.js/rewrites#path-matching)

We tried this change in rewrites method of `next.config`
```
{
        source: '/a/:account_id/s/:showcase_id*',
        destination: '/account/:account_id/showcase/:showcase_id*',
},
```

**Comment by icyJoseph** [comment]
Hi,

Try with something like:

```js
  async rewrites() {
    return [
      {
        source: "/t/:type(\\w+)/f/:feature(\\w+)*",
        destination: "/type/:type/feature/:feature*",
      },
    ];
  },
```

Of course, use proper names for your situation.

**Reply by icybergenome** [reply]
@icyJoseph I tried this:
```
{
        source: '/a/:account_id(\\w+)/s/:showcase_id(\\w+)*',
        destination: '/account/:account_id/showcase/:showcase_id*',
      },
```

and it didn't work

**Reply by icyJoseph** [reply]
It does work for me.

Could you clarify what is it you want to do, what are you expecting, and how does your file system look like?

**Reply by icyJoseph** [reply]
I have it like this:

```console
pages/account/
└── [account_id]
    └── showcase
        └── [showcase_id].tsx
```

And the next config file looks like:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return [
      {
        source: "/a/:account_id(\\w+)/s/:showcase_id(\\w+)*",
        destination: "/account/:account_id/showcase/:showcase_id*",
      },
    ];
  },
};

module.exports = nextConfig;

```

Then I when I go to: http://localhost:3000/a/foo/s/bar I see:

<img width="727" alt="Screenshot 2023-01-04 at 16 06 50" src="https://user-images.githubusercontent.com/21013447/210585133-c47eb332-1c46-4b25-ada0-d9b6e6b24c55.png">

Which are the contents of:

```tsx
// pages/account/[account_id]/showcase/[showcase_id].tsx
import { type NextPage } from "next";
import { useRouter } from "next/router";

const Feature: NextPage = () => {
  const { query } = useRouter();
  const { account_id, showcase_id } = query;

  return (
    <div>
      <h1>Feature Page</h1>
      <p>{account_id}</p>
      <p>{showcase_id}</p>
    </div>
  );
};

export default Feature;

```

**Reply by icybergenome** [reply]
In my case it lands me to 404 page

**Reply by icybergenome** [reply]
Apologies for the delay in response. In my case my structure looks like this:

pages/account/
└── [account_id]
    └── showcase
        └── [showcase_id]/
            └──  sharing.tsx

And I am trying the url: `http://localhost:3000/a/9fd16b7f-6388-4178-a622-e02434d7bf1a/s/7f806371-9330-449e-b1f9-3ab559fba66a/sharing`
